src/components/__tests__/NewTabThemeWrapperHOC.test.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import { flowRight } from 'lodash/util'
+import { showBackgroundImages } from 'src/utils/featureFlags'
+import NewTabThemeWrapper from '../NewTabThemeWrapperHOC'
+
+jest.mock('src/utils/featureFlags', () => ({
+  showBackgroundImages: jest.fn(),
+}))
+
+describe('Theme Wrapper HOC', () => {
+  it('renders without error', () => {
+    const DummyComponent = (props) => <div {...props} />
+    const WrappedHOC = flowRight([NewTabThemeWrapper])(DummyComponent)
+    expect(() => {
+      shallow(<WrappedHOC />)
+    }).not.toThrow()
+  })
+
+  it('renders with a white text palette when background image is enabled', () => {
+    showBackgroundImages.mockImplementation(() => true)
+    const DummyComponent = (props) => <div {...props} />
+    const WrappedHOC = flowRight([NewTabThemeWrapper])(DummyComponent)
+    const testComponent = shallow(<WrappedHOC />)
+    const themePassedIn = testComponent.props().theme
+    expect(themePassedIn.palette.backgroundContrastText.main).toEqual('#fff')
+  })
+
+  it('renders with a black text palette when background image is disabled', () => {
+    showBackgroundImages.mockImplementation(() => false)
+    const DummyComponent = (props) => <div {...props} />
+    const WrappedHOC = flowRight([NewTabThemeWrapper])(DummyComponent)
+    const testComponent = shallow(<WrappedHOC />)
+    const themePassedIn = testComponent.props().theme
+    // testing that
+    expect(themePassedIn.palette.text.backgroundContrastText).toEqual(undefined)
```

Comment:
@kmjennison ok so backgroundContrastText is not in the default theme palette that material UI exposes.  We add it at the high level theme wrapper of the app, but that context is not available in my dummy test component.  I tried wrapping everything in a theme provider and providing our default theme but then I was having a hard time getting the theme of the nested component.  I opted to just test that it's 'undefined' which is basically saying "make sure that our nested theme doesn't override the default theme on this value when enable background is disabled"

Reply:
thoughts?

Reply:
This sounds reasonable. Go ahead and just add this rationale in a comment.